### ISSUE TITLE
Definite integral error

### ISSUE THREAD

**Issue by onurtoker** [sympy/sympy#26071]
Sympy version 1.12, Python version 3.10.12 (Working on Google Colab)

Definite integral error

c0 = integrate((0.5 - t) * Heaviside(t-0.5), (t, 0, 0.5))

evaluates to 0.125, but must be zero.

**Comment by oscarbenjamin** [comment]
> The integrand function `(0.5 - t) * Heaviside(t-0.5)` is equal to `0.5` for t between `0 `and `0.5`.

No, it is equal to zero as you can see if you plot the function. `Heaviside(t-0.5)` is zero for `t - 0.5 < 0`.

**Comment by abhayku2002** [comment]
So in this case, `t-0.5` is less than `0` for all `t` in the interval `[0, 0.5]`, so the Heaviside function should indeed return `0`, making the integral 0. Am I correct?

**Comment by abhayku2002** [comment]
I think it can be solve like this 
```python
import sympy as sp

def my_heaviside(t):
    return sp.Piecewise((0, t < 0), (1, t > 0))

t = sp.Symbol('t')
c0 = sp.integrate((0.5 - t) * my_heaviside(t-0.5), (t, 0, 0.5))

print(c0)
```
Output: ` 0`

**Comment by oscarbenjamin** [comment]
Yes, the integral should be zero.

The problem is meijerg:
```python
In [14]: integrate((0.5 - t) * Heaviside(t-0.5), (t, 0, 0.5))
Out[14]: 0.125000000000000

In [15]: integrate((0.5 - t) * Heaviside(t-0.5), (t, 0, 0.5), meijerg=False)
Out[15]: 0
```
Either meijerg should not try to evaluate this integral or it should be fixed to handle this integral properly or perhaps this should be handled by a rewrite to Piecewise or something.
